File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int minp=Integer.MAX_VALUE;
        int maxpr=0;
        for(int i=0;i<n;i++){
            if(prices[i]<minp){
                minp=prices[i];
            }
            else if(prices[i]-minp>maxpr){
                maxpr=prices[i]-minp;
            }

        }
        return maxpr;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max=0; int profit=0; int buy=prices[0];
        int n= prices.size();
        for(int i=1;i<n;i++){
            profit=prices[i]-buy;
            if(profit<0){
                profit=0;
                buy=prices[i];
            }
            if(max<profit) max=profit;
        }
        return max;
    }
};